### PR TITLE
Feature/dyanmic paths

### PR DESCRIPTION
**Pull request by tomdavies**
I went down a _little bit_ of a rabbit hole poking around for a fix to #68

TLDR; currently we assume that the Craft `storage` dir is in the project root, and named `storage`. Most of the time that will hold, but not always. We also hard code that all over the place.

This PR refactors the way `copy/db/up` / `copy/db/down` work to read the storage path either direct from Craft (local) or from the stage config (remote stages). 

It also updates `copy/setup` to store the relevant paths in `StageConfig` after reading them from the stage by calling a new action (`PathsAction` / `copy/paths`)  during setup.

I suspect it could use some more work, but I learned a ton about the plugin's internals while I wrote it :)

### CODE REVIEW

**Comment by joshuabaker** [comment]
@tomdavies Thanks for this!

I’m curious about the difference between local and remote here. If the codebase is the same, I’d assume (perhaps incorrectly) that these would be the same path (related to the root). Is that not the case?
